homeassistant/components/bosch_shc/__init__.py

```diff
@@ -76,23 +79,21 @@ async def stop_polling(event):
         await hass.async_add_executor_job(session.stop_polling)
 
     await hass.async_add_executor_job(session.start_polling)
-    hass.data[DOMAIN][entry.entry_id][DATA_POLLING_HANDLER] = (
-        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_polling)
+    entry.runtime_data.polling_handler = hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_STOP, stop_polling
```

Comment:
why don't we use `entry.async_on_unload` here?

Reply:
> why don't we use `entry.async_on_unload` here?

Adjusted accordingly